packages/js/src/plugins/auctionHouseModule/operations/directSell.ts

```diff
@@ -90,7 +84,10 @@ export type DirectSellInput = {
    *
    * This includes, its asset, auction house address, buyer, receipt address etc.
    */
-  bid: Omit<Bid, 'bookkeeperAddress' | 'purchaseReceiptAddress' | 'createdAt'>;
+  bid: Omit<
+    PrivateBid,
```

Comment:
I still think we should allow "direct sell" to work on public bids. It's just that we need to ask the seller to provide the token account and default to the ATA.

To Recap:
- We accept a `bid` which can bid public or private and an optional `sellerTokenAccount` which defaults to the ATA between the seller and `bid.asset.address`.
- If `bid` is private, then we need to make sure the token accounts match.
- If `bid` is public, then we use the seller token account to figure out which asset to sell.
- Since we need to return the bid from the operation, we'll need to fetch the asset but we need to do that on the operation handler. Not on the transaction builder.

Reply:
Does it need it? If so can it be pushed to the operation handler instead of the transaction builder?

Reply:
Please, check again. I did a revisit to allow Listing to store assets without a token.

Reply:
That way it's only required to load up Purchase after the direct buy/sell

Reply:
It can be optimized not to load purchase when we have token provided, but I haven't done it yet.